### PR TITLE
Lic9, closes #14

### LINKED ISSUE #14: LIC 9
There exists at least one set of three data points separated by exactly C PTS and D PTS
consecutive intervening points, respectively, that form an angle such that:
angle < (PI − EPSILON)

or

angle > (PI + EPSILON)
The second point of the set of three points is always the vertex of the angle. If either the first
point or the last point (or both) coincide with the vertex, the angle is undefined and the LIC
is not satisfied by those three points. When NUMPOINTS < 5, the condition is not met.
1 ≤ C PTS, 1 ≤ D PTS
C PTS + D PTS ≤ NUMPOINTS − 3

### PR DESCRIPTION
**Pull request by a-kbd**
Tests included, which were kind of a pain to write and I'm not 100% satisfied that they're testing everything but it'll do

### CODE REVIEW

File: src/junit/LIC_test.java [hunk 1]
```diff
@@ -30,4 +31,36 @@ public void checkIfInvalidGivesFalse(){
         CMV cmv = new CMV(datapoints);
         assertFalse(cmv.lic5_calculate());
     }
+
+    @Test
+    public void test_lic9(){
```
Comment (andersblomqvist) [inline src/junit/LIC_test.java:36]: You should also test for when: 
* `C_PTS < 1`
* `D_PTS < 1`
* `C_PTS + D_PTS <= NUMPOINTS - 3`

File: src/main/CMV.java [hunk 2]
```diff
@@ -108,7 +110,41 @@ private Boolean lic8_calculate() {
...
+    private boolean lic9_calculate() {
+
+        //prevents array out of bounds (sum of points between and the two points themselves)
+        int points_bound = Parameters.C_PTS + Parameters.D_PTS + 2;
+        double pi = Math.PI;
+        double ep = Parameters.EPSILON;
+
+        if(datapoints.length < 5){
+            return false;
+        }
```
Comment (andersblomqvist) [inline src/main/CMV.java:122]: Add checks for when:

* `C_PTS < 1`
* `D_PTS < 1`
* `C_PTS + D_PTS > datapoints.length - 3`

File: src/junit/LIC_test.java (hunk 1 above)
Comment (andersblomqvist) [reply src/junit/LIC_test.java:36]: Additionally, explain in natural language what the test is doing. From [Canvas](https://canvas.kth.se/courses/37918/assignments/235345) it says:

* The contract that is tested is clearly formulated in natural language
* The assertions of the tests correctly and completely assess the contract

**Review by andersblomqvist — CHANGES_REQUESTED** [review]
Some minor changes needs to be made. Otherwise it looks good.

Regarding the global parameter values, they should be set accordingly in each test for making sure they are ok.

File: src/junit/LIC_test.java (hunk 1 above)
Comment (a-kbd) [reply src/junit/LIC_test.java:36]: Those are conditions on the parameters themselves, such parameters should never be passed to the function in the first place, I can't simply return false for those conditions as that is as arbitrary as returning true, it is just undefined

I can add a check that crashes the program if such parameters are passed (also the points_bound variable kind of ensures the same as the last condition ensures)

Regarding test comments, I'll add them

**Review by andersblomqvist — APPROVED** [review]
Good fix. Think the explanations are accurate.

**Pull request merged**